R/helpers.R

```diff
@@ -267,11 +250,81 @@ CreateVDJData <- function(new, orig = NULL) {
 #' @param n Colors in spectrum. Default = 100
 
 ColorScale <- function(name = c("coolwarm", "viridis"), n = 100) {
-    if (is.null(name) || name == "coolwarm") {
+    if (is.null(name) || (name %in% c("coolwarm","DALI","DALII","Pastel","Spectrum"))) {
         return(colorRampPalette(c("#4575B4", "#91BFDB", "#E0F3F8", "#FFFFBF", "#FEE090", "#FC8D59", "#D73027"))(n))
-    } else if (name == "viridis") {
+    } else if (name == "viridis" | name == "Colorblind") {
         return(colorRampPalette(c("#440154", "#443A83", "#31688E", "#21908C", "#35B779", "#8FD744", "#FDE725"))(n))
     } else {
         stop("invalid colorscheme ", name)
     }
 }
+
+#' Returns colorvector based on chosen theme and amount of categories
+#'
+#' @param data data to get categories from
+#' @param theme coloscheme to use. Default = DALI
+#'
+#' @importFrom Polychrome sky.colors
+#'
+#' @export
+
+GetCategoricalColorPalette <- function(data, theme = "DALI") {
+    if (!is.null(theme)) {
+        if (length(theme) > 1) {
+            return(theme)
+        }
+    }
```

Comment:
Shouldn't this stop the function instead of returning the input theme?

Reply:
After looking at plotting.R, I understand what this is supposed to do.
It still doesn't feel right to me to add this here. If you already have a colorpalette, you shouldn't be calling this function to just return what you already have.